jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
This is turning into a long discussion, but the topic is important as we really need to start pushing SimPhoNy to the "next level".

Let us assume that there will be an ```SimPhoNyApplication``` class which will carry references to SP,SD,BC, and CM objects. Then the engine/wrapper interface will (probably) include a method like ```set/configure``` for uploading the application into the wrapper with a single call. Inside this method, the wrapper will pick/search for all the parameters/information from the application object which are deemed necessary for executing the run method; also the appropriate ```proxy datasets``` are created in this method and, subsequently, maintained by the wrapper.

Then, for example, a particular question arises (there are many): should the wrapper interface still include the method ```add_dataset```?